### PR TITLE
measurement_rsp: remove hardcoded measurement block

### PR DESCRIPTION
**Pull request by longlongyang**
removed hardcoded measurement block in measurement_rsp.rs

Signed-off-by: Longlong Yang <longlong.yang@intel.com>

### CODE REVIEW

File: spdmlib/src/message/measurement.rs [hunk 1]
```diff
@@ -98,7 +99,15 @@ pub struct SpdmMeasurementsResponsePayload {
 
 impl SpdmCodec for SpdmMeasurementsResponsePayload {
     fn spdm_encode(&self, context: &mut common::SpdmContext, bytes: &mut Writer) {
-        self.number_of_measurement.encode(bytes); // param1
+        //When Param2 in the requested measurement operation is 0 , this
+        //parameter shall return the total number of measurement indices on
+        //the device. Otherwise, this field is reserved.
+        if self.measurement_record.number_of_blocks == 0 {
+            // SpdmMeasurementOperation::SpdmMeasurementQueryTotalNumber
+            (MAX_SPDM_MEASUREMENT_BLOCK_COUNT as u8).encode(bytes); // param1
```
Comment (jyao1) [inline spdmlib/src/message/measurement.rs:107]: This is wrong.
We cannot use MAX_SPDM_MEASUREMENT_BLOCK_COUNT.
We must use real measurement block count from device security lib.

Comment (longlongyang) [reply spdmlib/src/message/measurement.rs:107]: Fine, But I have no idea where could I get the value from. Or just use 5 now?

Comment (jyao1) [reply spdmlib/src/message/measurement.rs]: spdmlib should get it from below function. No hardcode is allowed in spdmlib.

type SpdmMeasurementCollectionCbType = fn(
    spdm_version: SpdmVersion,
    measurement_specification: SpdmMeasurementSpecification,
    measurement_hash_algo: SpdmBaseHashAlgo,
    measurement_index: usize,
) -> Option<SpdmMeasurementRecordStructure>;

File: spdmlib/src/message/measurement.rs [hunk 2]
```diff
@@ -98,7 +98,15 @@ pub struct SpdmMeasurementsResponsePayload {
 
 impl SpdmCodec for SpdmMeasurementsResponsePayload {
     fn spdm_encode(&self, context: &mut common::SpdmContext, bytes: &mut Writer) {
-        self.number_of_measurement.encode(bytes); // param1
+        //When Param2 in the requested measurement operation is 0 , this
+        //parameter shall return the total number of measurement indices on
+        //the device. Otherwise, this field is reserved.
+        if self.measurement_record.number_of_blocks == 0 {
+            // SpdmMeasurementOperation::SpdmMeasurementQueryTotalNumber
+            5_u8.encode(bytes); // param1
```
Comment (jyao1) [inline spdmlib/src/message/measurement.rs:106]: It is still wrong.
See my previous feedback.

Comment (longlongyang) [reply spdmlib/src/message/measurement.rs:106]: Got it, will re-submit. Thank you!

**Pull request merged**